onnxruntime/core/providers/mkldnn/mkldnn_execution_provider.h

```diff
@@ -4,11 +4,18 @@
 #pragma once
 
 #include <memory>
-
+#include <map>
+#include <memory.h>
 #include "core/framework/allocatormgr.h"
 #include "core/framework/execution_provider.h"
 #include "core/graph/graph_transformer.h"
 
+#ifndef MKLDNN_HPP
```

Comment:
do we need this?

Reply:
Yes. We are forward defining `mkldnn::memory` `py_byild` (`--enable_pybind`) was failing. 
When I include this in `conv.h`, it was giving compilation error as it also has `mkldnn.hpp` included.

Reply:
i understand that you're forward defining 'memory'. i didn't understand the ifndef part. why is that needed?

Reply:
conv.cc gets mkldnn::memory definition from both mkldnn_provider.h and mkldnn.hpp and we get compilation error. ifndef MKLDNN_HPP makes conv.cc ignore forward definition from mkldnn_provider.h

Reply:
will look into this more. sounds like a hack.

Reply:
Removed #ifndef.